colossalai/inference/core/engine.py

```diff
@@ -97,21 +97,29 @@ def __init__(
             self.capture_model(self.k_cache, self.v_cache)
 
         # Model and relatable attrs of speculative decoding will be set by `enable_spec_dec`
-        self.use_spec_dec = False
+        self.use_spec_dec = self.inference_config.use_spec_dec
+
+        # TODO: when use_spec_dec set to True, users should pass drafter_model configs into InferenceEngine
+        #    We can add a SpecDecConfig class to store these configs.
```

Comment:
Just to notify: Drafter related configs are set in `InferenceConfig` and the drafter model is passed in `enable_spec_dec`. I noticed that you didn't modify the logic of enabling and disabling speculative decoding